fusor-ember-cli/app/routes/review/installation.js

```diff
@@ -121,8 +121,43 @@ export default Ember.Route.extend(NeedsExistingManifestHelpers, {
       },
       data: {}
     }).then(response => {
-      controller.set('validationErrors', response.validation.errors);
-      controller.set('validationWarnings', response.validation.warnings);
+       if (Ember.isPresent(response.validation)) {
+         controller.set('validationErrors', response.validation.errors);
+         controller.set('validationWarnings', response.validation.warnings);
+       }
+    })
+  },
+
+  validateDiscoveredHostsPoweredOn() {
+    let controller = this.get('controller');
+    let deployment = this.get('controller.model');
+    let token = Ember.$('meta[name="csrf-token"]').attr('content');
+    if (!deployment.get('deploy_rhev') || Ember.isPresent(controller.get('validationErrors'))) {
+      return Ember.RSVP.Promise.resolve('no validateDiscoveredHostsPoweredOn needed');
+    }
+
+    deployment.get('discovered_hosts').then(function(results) {
+      var discoveredHostIds = results.getEach('id');
+      discoveredHostIds.push(deployment.get('rhev_engine_host_id'));
+      controller.set('showSpinner', true);
+      controller.set('spinnerTextMessage', controller.get('spinnerTextMessage') + " and ensuring discovered hosts are powered on");
+      discoveredHostIds.forEach(function (hostId) {
+        request({
+          url: `/api/v2/discovered_hosts/${hostId}/refresh_facts`,
+          type: "PUT",
+          headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "X-CSRF-Token": token
+          },
+          data: {}
+        }).catch(error => {
```

Comment:
I can see why you were catching here because the catch on the promise chain doesn't fit.  However, the actual jqXHR request was successful in other cases.

It might be better to restructure this to return a new pormise, catch the error, set validationErrors (which we check for later), and resolve.